main.tf

```
@@ -63,7 +63,7 @@ resource "aws_db_instance" "default" {
 }
 
 resource "aws_db_subnet_group" "default" {
-  count      = "${local.enabled ? 1 : 0}"
+  count      = "${local.enabled && !var.same_region ? 1 : 0}"
```

```suggestion
  count      = "${local.enabled && var.same_region == "false" ? 1 : 0}"
```

Since `var.same_region` is a string, not boolean